crypto/asn1/ameth_lib.c

```diff
@@ -71,33 +71,36 @@ static const EVP_PKEY_ASN1_METHOD *pkey_asn1_find(int type)
 }
 
 /*
- * Find an implementation of an ASN1 algorithm. If 'pe' is not NULL also
- * search through engines and set *pe to a functional reference to the engine
+ * Find an implementation of an ASN1 algorithm. Also search through engines.
+ * If 'pe' is not NULL set *pe to a functional reference of the engine
  * implementing 'type' or NULL if no engine implements it.
  */
 
 const EVP_PKEY_ASN1_METHOD *EVP_PKEY_asn1_find(ENGINE **pe, int type)
 {
     const EVP_PKEY_ASN1_METHOD *t;
+#ifndef OPENSSL_NO_ENGINE
+    ENGINE *e;
+#endif
 
     for (;;) {
         t = pkey_asn1_find(type);
         if (!t || !(t->pkey_flags & ASN1_PKEY_ALIAS))
             break;
         type = t->pkey_base_id;
     }
-    if (pe) {
+
 #ifndef OPENSSL_NO_ENGINE
-        ENGINE *e;
-        /* type will contain the final unaliased type */
-        e = ENGINE_get_pkey_asn1_meth_engine(type);
-        if (e) {
+    /* type will contain the final unaliased type */
+    e = ENGINE_get_pkey_asn1_meth_engine(type);
+    if (e) {
```

Comment:
Please note that our coding style mandates `if (e != NULL)` for NULL pointer checks.

_(I know it was this way already, but please fix it while you're at it)_

Reply:
OH!  For sure!  `!= NULL` has always been my preference, but I put pre-existing style conformance as the top priority.  Will happily make these changes soon.